internal/service/document/document_rerun.go

```
@@ -0,0 +1,120 @@
+
+	"gorm.io/gorm"
+
+	"ragflow/internal/dao"
+	"ragflow/internal/entity"
+)
+
+// ErrRerunDocumentNotFound is the rerun endpoint's only "unknown or denied"
+// answer: an unknown log id, a missing document/dataset, and an access
+// denial all collapse to the same message so a caller cannot probe whether
+// a document exists in another tenant.
+var ErrRerunDocumentNotFound = errors.New("Document not found.")
```

golangci-lint (2.13.2)

[error] ST1005: error strings should not end with punctuation or newlines

(staticcheck)